pkg/util/mount/mount.go

```diff
@@ -286,7 +286,7 @@ func IsNotMountPoint(mounter Interface, file string) (bool, error) {
 // use in case of bind mount, due to the fact that bind mount doesn't respect mount options.
 // The list equals:
 //   options - 'bind' + 'remount' (no duplicate)
-func isBind(options []string) (bool, []string) {
+func isBind(options []string) (bool, []string, []string) {
```

Comment:
Why is this being special cased in the bind mount? Shouldn't it be a mount option for the global mount? That way the bind mounts can be simple?

Reply:
The problem is - when we bind mount we are dropping *ALL* mount options supplied by global mount. That is because traditionally Kernel drops most mount options when bind mounting something.  So even if global mount did use `_netdev`, the bind mount will not use it.  

So, this fix makes sure that - if global mount had `_netdev`, then we must carry that option when creating bind mount. We ourselves don't supply `_netdev` but we expect that mount option to come from global mount (either via storageClass mount option or plugin uses `_netdev` as one of the default mount options).

Reply:
can we just apply all mount options we applied to global mount to the bind mount too? Then we don't have to special case netdev

Reply:
It's a bit confusing because it looks like for `bindRemountOpts` we are applying all the mount options from the global mount.

Reply:
The problem is - remount also does not copy `_netdev` mount option over, if original bind mount did not had it. I am somewhat skeptical about using remount to apply mount options to be honest. We know that - mount options that Kernel knows about are automatically copied over from original mount point when bind mount is created. We don't need remount for that.

And userspace mount options aren't copied over on remount.

Reply:
Where is the "_netdev" mount option coming from since it's not from the global mount?

I see that:
1. bind options will return "bind" and "_netdev" if it's set
2. bind remount options will return "bind", "remount", and all other options (including "_netdev")

Do we really need to special case the "_netdev" in the first bind mount if the remount is going to already include it?

Reply:
> Do we really need to special case the "_netdev" in the first bind mount if the remount is going to already include it?

Yeah we do need to. If first bind mount did not include `_netdev` mount option then `_netdev` mount option on remount has no effect and it isn't applied.

Reply:
Thanks for clarifying.  This looks fine then.  Another issue I found is that all other plugins besides rbd are not using storageclass mount options in SetUp, so this fix will only work for rbd.  We can handle the other plugins separately.

Reply:
Yeah - I logged https://github.com/kubernetes/kubernetes/issues/68797 to fix that as a follow up item.